### PR TITLE
[Compose] Create TrainOfIcons reusable Composable

### LINKED ISSUE #18583: [Jetpack Social] Create social icons "train" component in Compose
Jetpack Social improvements will include a UI element displaying the supported social network icons arranged in a row, which looks like the existing `TrainOfAvatars` UI we use in Reader screen and Prompts card.

This is an opportunity to create this reusable element in Compose, so we can use it in future UIs, starting with the Jetpack Social UI for the "train" of social icons.

Figma: Th1ahHKq53k5JT1PNDMavY-fi-865_13166

Image

### PR DESCRIPTION
**Pull request by thomashorta**
Fixes #18583

This is similar to the `TrainOfAvatars` Android view but more flexible. It is being created for use in the Jetpack Social improvements, which contain a "train" of social network icons, but we can use it to replace the `TrainOfAvatars` view in other future Compose layouts.

This uses `AsyncImage` to load the icons/avatars/images so it's flexible in terms of the image model / source.

image

_Note: the image content is not properly loaded in the Preview because `AsyncImage` is being used, which doesn't load in Android Studio's preview, but it's possible to run the Preview on a device to see it._

image

## To test
Nothing to test as this creates the Composable but does not use it at the moment. It's possible to sync the code and check the Compose previews and run the component preview on a device from there.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] RTL

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt]: np: I guess we should add contentDescription and iconBorderWidth here; wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt:77]: Is duplicating the `.clip` needed here?

**Review by develric — COMMENTED** [review]
Thanks @thomashorta for creating the component, and thanks @RenanLukas for taking over while Thomas is AFK 🙇 

I tested the previews on a device also checking the talkback against the contentDescription and it works well; also RTL works as expected 👍 

I left a couple of comments for your review before to final approve. Let me know 🙇

File: WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt:77]: Good point! We actually need them both.

If we remove just the first `.clip`, it looks like this:
image

And if we remove just the second one, it looks like this:
image

Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt:77]: Makes sense 👍 , thanks for elaborating 🙇

**Review by develric — APPROVED** [review]
Thanks @RenanLukas for adding the missing params documentation, and double checking the rationale of the double .clip

LGTM. I'm approving the PR, feel free to merge once the CI is passing 👍

File: WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt:77]: Just adding more context for what is happening there. The modifier operations run in the order they're called. So, assuming the icon size is `32dp`, and border size is `2dp`, what's happening there is:

1. `.size(iconSizeWithBorder.dp)` -> makes the component have size `iconSizeWithBorder`, or `36dp`;
2. `.clip(CircleShape)` -> clip the component to a circle shape, so anything drawn outside of that circle will not show up;
3. `.background(MaterialTheme.colors.surface)` -> color the component background with the `surface` color (e.g.: white for light theme);
4. `.padding(iconBorderWidth)` -> add padding on all sides of size `iconBorderWidth`, or `2dp`, so any content drawn by the component (`AsyncImage`) will be inset on top of the circular `surface` colored background (because of the previous steps), so if the content is opaque it will look like a `2dp` border;
5. `.clip(CircleShape)` finally clips the content that will be drawn, so the loaded image also has a circular shape, and because of the padding on all sides, the size of this circle will be `32dp` (36 - 2 - 2).

This is why things look like what @RenanLukas showed above:
- If we remove the first `.clip`, then the outside shape will be a square, with `2dp` padding, with a circle image inside.
- If we remove the second `.clip`, then the outside shape will be a circle, with `2dp` padding, with a square image inside.

**Pull request merged**
